packages/gatsby-telemetry/src/store.ts

```diff
@@ -0,0 +1,72 @@
+import path from "path"
+import {
+  appendFileSync,
+  readFileSync,
+  renameSync,
+  readdirSync,
+  existsSync,
+  unlinkSync,
+} from "fs"
+
+export class Store {
+  bufferFilePath: string
+  baseDir: string
+  eventsJsonFileName = `events.json`
+
+  constructor(baseDir: string) {
+    this.bufferFilePath = path.join(baseDir, this.eventsJsonFileName)
+    this.baseDir = baseDir
+  }
+
+  appendToBuffer(event: unknown): void {
+    try {
+      appendFileSync(this.bufferFilePath, event, `utf8`)
+    } catch (e) {
+      //ignore
+    }
+  }
+
+  async flushFile(
```

Comment:
If these are async functions, why do we need to use sync fs operations? Wouldn't it be better to use promisified versions from fs-extra and avoid the blocking i/o?

Reply:
This always happens in a background process that does nothing else so blocking io shouldn't impact